### PR TITLE
CopterHack 2023: Swarm in Blocks 2 (Atena)

### PR DESCRIPTION
**Pull request by joseCarlosAndrade**

### CODE REVIEW

**Comment by SeliverstovaE** [comment]
Dear friend,

Thank you for your interest in the CopterHack 2023 contest. Your application has successfully passed the experts’ review and was accepted for the second project stage of the competition, which will be held until February 28, 2023. This stage will be followed by monthly updates, lectures, and guidance from industry experts.

What's next?

1. Join the Telegram group with participants of the second project stage “[CopterHack 2023, participants chat](https://t.me/+t8hLtf-w2EwzOWM6)”.  
2. Every month, you're expected to prepare free-form reports on the progress of your project (update) in this draft pull request. In your report please indicate your progress, what tasks have been solved. You are encouraged to attach photos and videos that demonstrate the results of the work done within the month.
3. Send the link to the update to the participants chat in the Telegram group.

In response to the message, you will receive feedback from the competition jury.

There is no specific date for sending the report, so you can send the report between the beginning and the end of the month.

Monthly updates become mandatory from November 2022.

We are interested in helping you to complete your project as much as possible by the final stage. We wish you success!

Always in touch! Your CopterHack team.

**Comment by joseCarlosAndrade** [comment]
(#November #Atena)
##November update
This month, we focused on introducing the Clover platform to our new members, as well as our previous project Swarm In Blocks. We assigned trainings and helped them so we can start developing more specific parts of our project on the following months. In order to get used to clover and ROS platform, we've done meetings and discussion topics on the project objetives and struggles with Atena members that participated in CopterHack 2022.

**Comment by pteacher** [comment]
Cool work! Well done!

**Comment by joseCarlosAndrade** [comment]
(#November #Atena)
##December update

We've gone through meetings to discuss the new features that the Swarm in Blocks 2 project must have. We decided to divide the development in different areas, such as Swarm Features and Physical Drone Flying to better organize ourselves. Our new members are in the process of getting to install and know better these platforms, as well as the ROS environment.

**Comment by joseCarlosAndrade** [comment]
(#December #Atena)
##December update

We've gone through meetings to discuss the new features that the Swarm in Blocks 2 project must have. We decided to divide the development in different areas, such as Swarm Features and Physical Drone Flying to better organize ourselves. Our new members are in the process of getting to install and know better these platforms, as well as the ROS environment.

**Comment by joseCarlosAndrade** [comment]
(#January #Atena)
##January update

In January we effectively started our project development. We set off with some of the features that will be added to our platform Swarm in Blocks, most of them based on web applications so far, being: 
- Web Firsts Person Viewer - An easier way to take control of a single drone having the first person perspective and giving navigation commands through a computer keyboard (in a web interface, so it doesn't rely on desktop interfaces). This feature is in its finishing stage, with the back-end already functional and only some remaining tweaks focused on the front-end to be implemented;
- Web terminal - An easier and more accessible way of pulling out a terminal for fast commands and ROS / clover resolutions, but coming directly from our interface. This feature is also in its finishing stage and we're currently planning the integration with our platform, in addition to the front-end touches;

(It's possible to see those features and the rest of the visual update on the linked video in the telegram message for this update!).

We also started working on the new homepage for our application and the new ground station platform for real swarm manipulation and control. We're currently in the process of discussing and planning what would be interesting to have in this GS so it makes swarm controlling easier and more accessible (using our swarm programming blocks).

More features are in the process of development, focusing on providing ways of extracting crucial real-time information about the swarm to allow real flight safety and control. Furthermore, we're thinking of solutions to enable real-time communication between each clover and the master (from where the commands will be given) and the necessary network architecture to make all of this possible.

We're also working on the positioning system for multiple drones (currently with arucos) to allow indoor swarm flights (in controlled environments).

**Comment by joseCarlosAndrade** [comment]
(#February #Atena)
##February update

In February, our team really focused on the actual development of our project, mainly on the coding parts. We built the swarm station, which is a ground control station intended to facilitate the use and control of clover swarms. We also developed our homepage, to increase user experience when navigating through our project. We'll put some screenshots here from our work but we'll release a video showing a bit of everything we've been developing.


About the mentioned features:
- **Swarm station**
swarm_station
This is our feature to control and have access to real-time information about the swarm, besides having the power to give emergency instructions. It's built using the ROS3Dlib javascript library for 3D ROS marker plotting, and we're running its backend with a python node that handles the position and marker mesh for each clover, dynamically. One of its advantages is that it runs totally on the web browser, making it a lot more practical. For the next month, we intend to improve the visualization, since we think it's a bit too dark to actually see each clover.

- **Our new homepage**
WhatsApp Image 2023-02-28 at 17 56 01
WhatsApp Image 2023-02-28 at 17 56 10
WhatsApp Image 2023-02-28 at 17 56 21
WhatsApp Image 2023-02-28 at 17 56 30
WhatsApp Image 2023-02-28 at 17 56 39

This is our new homepage, built to provide a better user experience for the swarm manipulation task.

- **Web Terminal**
terminal_web
This feature is intended to give the user quick access to terminal emulators so they can have better control of their work. It instantiates a new terminal backend using node.js each time you click the button. Now we're planning on integrating this feature into the Swarm Station. We intend to make a pop-up every time the user clicks on the terminal button, and it can generate several terminals instantly. 

- **FPV**
fpv
The First Person Viewer is also built 100% on the web browser, making it a lot more practical than our original python WX version. It mostly works for every case, but there're a few tweaks we're planning on doing for the next month, in order to make it more robust.

- **Real Swarm**
About our most audacious task, which is to fly a real swarm with real clovers, we started to plan the necessary architecture to make it real. We're planning on running them on the same ROS_MASTER_URI address, but for that, we have to put our work from last year's CopterHack to real life, which is renaming and remapping all the nodes, topics, services, and frames necessary to make a single drone fly without conflicting with other devices, because they'll all be on the same network. For the next month, our most important task is to make the clover swarm possible in real life. Our members already started to work on the local network and namespaces remapping, and next month we'll be able to present the videos flying an indoor swarm with arucos.

**Comment by okalachev** [comment]
Hi! A lot of asset files in you articles are way too large to add them into the repo. Note, that all the repo content goes to the Clover's RPi image, so an asset size limit is about 300-500 KB.

There are two options to resolve this issue:

* Reduce the images sizes and quality dramatically to fit the limits (I can do this for you).
* Host the images somewhere else (for example, in another branch on your forked repo) and insert full URLs to them. Note, that these images won't be accessible to the reader, when the article is viewed on a Clover.

File: docs/en/swarm_in_blocks_2.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+### What's new
+
+
+As already mentioned, of the various problems that can increase the complexity of a real swarm, we decided to deal with the ones that most afflicted us and reintegrated our solutions into our central platform, building a single extremely complete and cohesive platform.
+
+| Problem | Our Solution |
+| -------- | -------- |
+| Possible collision between drones (lack of safety especially for large Swarms)  | Collision Avoidance System  |
+| Giant clutter to keep track of all Clovers in a swarm individually (several terminals, many simulateneo computers with several people to keep track of)  | Swarm Station  |
```
Comment (okalachev) [inline docs/en/swarm_in_blocks_2.md:73]: >simulateneo ?

**Review by okalachev — CHANGES_REQUESTED** [review]

**Comment by joseCarlosAndrade** [comment]
Hi! Hm, we're really sorry. We're gonna fix it by updating the assets source to retrieve the files from our own repository. We're also going to correct the pointed spelling error. Thank you very much.

**Review by okalachev — APPROVED** [review]

**Pull request merged**